internal/naming/main.go

```diff
@@ -125,6 +125,16 @@ func Route(otelcol string, prefix string) string {
 	return DNSName(Truncate("%s-%s-route", 63, prefix, otelcol))
 }
 
+// ClusterRole builds the cluster role name based on the instance.
+func ClusterRole(otelcol string, namespace string) string {
+	return DNSName(Truncate("%s-%s-cluster-role", 63, otelcol, namespace))
+}
+
+// ClusterRoleBinding builds the cluster role binding name based on the instance.
+func ClusterRoleBinding(otelcol string) string {
+	return DNSName(Truncate("%s-cluster-role-binding", 63, otelcol))
+}
+
```

Comment:
I think just `%s-collector` would be better here. That's how we name most other resources, and I don't see any reason to embed the type in the name here.